### PR TITLE
Add re.iosphe.kcc

### PR DESCRIPTION
**Pull request by TingPing**

### CODE REVIEW

**Comment by TingPing** [comment]
bot, build re.iosphe.kcc

**Comment by TingPing** [comment]
Should we mark this as proprietary since it uses KindleGen? I could also patch the app to tell the user to download it to `~/.var/app/...` (it tells them `/usr/local/` currently) perhaps but that is kinda a lame UX.

File: re.iosphe.kcc.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      - type: file
+        path: re.iosphe.kcc.desktop
+      - type: script
+        dest-filename: apply_extra
+        commands:
+          - tar -xf kindlegen.tar.gz
+          - install -D kindlegen /app/extra/bin/kindlegen
+          - rm EULA* KindleGen* kindlegen* manual.html
+      - type: extra-data
+        only-arches: [i386, x86_64]
```
Comment (barthalion) [inline re.iosphe.kcc.yml]: Given it won't work on arm, you should also add flathub.json.

**Comment by barthalion** [comment]
Is there a way to set multiple licenses in appdata? cc @hughsie

**Comment by hughsie** [comment]
> Is there a way to set multiple licenses in appdata

Sure, AppData uses the SPDX grammar, so you can do things like "GPLv-2.0 and LGPL-2.0"

File: re.iosphe.kcc.yml (hunk 1 above)
Comment (TingPing) [reply re.iosphe.kcc.yml]: Everything but that blob should work fine on arm.

**Comment by TingPing** [comment]
@AcidWeb Do you have any interest in supporting this as an official package?

As a brief pitch for Flatpak it allows a single package to be ran on a wide range of distros in a self-contained environment that is easy to test and support. Since you currently only support 3 releases of Ubuntu (not even the latest one) it seems to be a valuable problem for you to solve.

**Comment by AcidWeb** [comment]
I'm already maintaining 6 separate binary pipelines (not only for Linux) for this project. So I'm open for suggestions.

Be aware that KindleGen require to accept license agreement **before** downloading. I also suggest to check dev branch. 7z + (7z rar module) now replaced all other archiving software.

**Comment by TingPing** [comment]
> I'm open to making this my only official Linux release.

Great! There are a few minor things to resolve upstream like adding the appstream file, fixing the desktop file (absolute paths..), renaming them to be an app-id format, etc.

> Be aware that KindleGen require to accept separate license agreement before downloading. Additionally I'm quite sure that license don't allow redistribution.

Right, so currently this package is using an "extra-data" feature of `flatpak` which downloads this binary when the user installs it so *we* aren't redistributing it. It doesn't have an interface to show a EULA though. Ideally I would like this to just be done by the application.

> I also suggest to check dev branch. 7z (+ 7z rar module) now replaced all other archiving software.

I did see that and it would be nice to get rid of `unrar` but for now Flathub only hosts stable releases.

**Comment by AcidWeb** [comment]
Feel free to provide PR against `dev` branch with changes you need. When everything we be in place and I will be sure that entire application is intact we can think about publishing it here.

**Comment by AcidWeb** [comment]
I will add some warning about KindleGen license directly in application.

**Pull request closed without merging**